app/views/alchemy/admin/pictures/_filter_and_size_bar.html.erb

```diff
@@ -1,27 +1,20 @@
 <div id="image_assign_filter_and_image_sizing">
-  <div class="button_with_label">
-    <%= link_to_dialog(
-      render_icon('upload'),
-      alchemy.new_admin_picture_path(
-        element_id: @element.blank? ? nil : @element.id,
-        content_id: @content.blank? ? nil : @content.id,
-        swap: @swap,
-        page: params[:page],
-        size: @size,
-        options: @options.to_json
-      ),
-      {
-        size: "540x550",
-        title: Alchemy.t(:upload_image)
-      },
-      {
-        class: 'icon_button',
-        title: Alchemy.t(:upload_image)
-      }
-    ) %>
-    <label><%= Alchemy.t(:upload_image) %></label>
-  </div>
-  <div class="toolbar_spacer"></div>
+  <% if can? :create, Alchemy::Picture %>
+    <%= render 'alchemy/admin/partials/upload_button',
+      label: Alchemy.t(:upload_image),
+      file_types: configuration(:uploader)['allowed_filetypes']['pictures'],
+      dropzone: '#assign_image_list',
+      form_object: Alchemy::Picture.new,
+      file_attribute: 'image_file',
+      while_assigning: true,
+      redirect_url: alchemy.admin_pictures_path(
+        size: params[:size],
+        filter: 'last_upload',
+        content_id: @content.try(:id),
+        element_id: @element.try(:id)
```

Comment:
Thought a lot about removing them, but as you see in the partial, they are all necessary. And the `redirect_url` here is current API of the pictures controller. We could reduce the amount of parameters needed, but that is out of the scope of this PR